src/main/java/seedu/address/model/task/Recurring.java

```diff
@@ -0,0 +1,140 @@
+import seedu.address.model.tag.Tag;
+import seedu.address.storage.Storage;
+
+public class Recurring {
+   private final RecurType type;
+   private final LocalDateTime referenceDateTime;
+
+   public static final String MESSAGE_CONSTRAINTS = "Recurring should be in the format d or w, for eg: rec/d";
+   public static final String MESSAGE_RECURRING_TASK_SUCCESS = "Recurring Task: %1$s";
+
+   public static final String VALIDATION_REGEX = "[dw]"; 
+   public static final DateTimeFormatter stringFormatter =
+            DateTimeFormatter.ofPattern("dd/MM/yy@HH:mm");
+
+   public Recurring(String recurringStringStorage) throws ParseException {
+        String recurTypeString = recurringStringStorage.substring(0,1);
+        String dateTimeString = recurringStringStorage.substring(1);
+        this.type = parseRecurType(recurTypeString);
+        this.referenceDateTime = parseDateTime(dateTimeString);
+   }
+
+   public Recurring(String recurringString, LocalDateTime referenceDateTime) throws ParseException{
+       this.type = parseRecurType(recurringString);
+       this.referenceDateTime = referenceDateTime;
+   }
+
+   public LocalDateTime parseDateTime(String dateTimeString) {
+       return stringFormatter.parse(dateTimeString, LocalDateTime::from);
+   }
+
+   public RecurType parseRecurType(String recurringString) throws ParseException {
+       if (recurringString.equals("d")) {
+           return RecurType.DAILY;
+       } else if (recurringString.equals("w")) {
+           return RecurType.WEEKLY;
+       } else {
+           throw new ParseException(Recurring.MESSAGE_CONSTRAINTS);
+       }
+   }
+   
+   /**
+    * Returns a copy of the task that is being reset with everything being the same except for whether the method is Done
+    * @param taskToReset
+    * @return copied task with done set to undone
+    */
+   public Task resetDone(Task taskToReset) {
+       assert taskToReset != null;
+
+        Name updatedName = taskToReset.getName();
+        Priority updatedPriority = taskToReset.getPriority();
+        Description updatedDescription = taskToReset.getDescription();
+        Set<Tag> updatedTags = taskToReset.getTags();
+        Optional<Reminder> sameOptReminder = taskToReset.getOptionalReminder();
+        Optional<Recurring> sameOptRecurring = taskToReset.getOptionalRecurring();
+        return new Task(
+                updatedName, updatedPriority, updatedDescription, new Done("N"), updatedTags, sameOptReminder, sameOptRecurring);
+   }
+
+   /** Returns true if a given string is a valid name. */
+   public static boolean isValidRecurring(String test) {
+       return test.matches(VALIDATION_REGEX);
+   }
+   
+   /**
+    * Returns Daily or Weekly for display on the card.
+    */
+   public String displayRecurring() {
+       String result = type.name().substring(0, 1) + type.name().substring(1).toLowerCase();
+       return result;
+   } 
+
+
+   public TimerTask generateTimerTask(Model model, Task taskToReset) {    
```

Comment:
maybe a better way of doing this is passing it Logic, then we initialize a command called Recurr command and make Logic execute it. this handles storage and we can also provide some user feedback when a recur happens. Not sure how to transfer the CommandResult yet to MainWindow, but can think of that later